R/data.table.R

```diff
@@ -1739,6 +1739,9 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
         thisEnv = new.env()  # not parent=parent.frame() so that gsum is found
         for (ii in ansvars) assign(ii, x[[ii]], thisEnv)
         assign(".N", len__, thisEnv) # For #5760
+        #fix for #1683
+        if (".I" %chin% all.vars(jsub)) 
+          assign(".I", if is.null(irows) seq_len(nrow(x)) else irows, thisEnv)
```

Comment:
@arunsrinivasan you were right. Added a test with subsetting to be sure.

The problem now is that

`DT[ , .(I[1L], V2[1L]), by = V1]` runs with `GForce`, but `DT[ , .I[1L], by = V1]` does not.